Source/Modules/javascript.cxx

```diff
@@ -2695,6 +2696,10 @@ int TypeScriptTypes::top(Node *n) {
      String *moduleNameKebabCase = Swig_string_kebabcase(Getattr(n, "name"));
      String *typesFilePath = NewStringf("%s%s-types.d.ts", SWIG_output_directory(), moduleNameKebabCase);
      declarationFilePtr = NewFile(typesFilePath, "w", SWIG_output_files());
+     if ( !declarationFilePtr ) {
+       Printf(stderr, "Error opening file (%s) : %s\n", typesFilePath, strerror(errno));
+	     SWIG_exit(-1);
```

Comment:
There is some strange indentation here, please check for the mix of tabs and spaces.

I'd also amend the error message to say "Error opening file \"%s\" for output: ..." as it could be useful if a file exists but is read-only, for example.